core/src/avm2/class.rs

```diff
@@ -78,7 +78,7 @@ pub struct Class<'gc> {
     name: QName<'gc>,
 
     /// The type parameter for this class (only supported for Vector)
-    param: Option<GcCell<'gc, Class<'gc>>>,
+    param: Option<Option<GcCell<'gc, Class<'gc>>>>,
```

Comment:
The double `Option` is pretty confusing here; I'd suggest adding a dedicated type, e.g.:
```rs
pub enum TypeParam<'gc> {
    Any,
    Class(GcCell<'gc, Class<'gc>>),
}

impl Eq for TypeParam<'gc> { ... }
```